Modules/Scripted/SegmentStatistics/SegmentStatistics.py

```diff
@@ -423,6 +423,17 @@ def computeStatistics(self):
 
     segmentationNode = slicer.mrmlScene.GetNodeByID(self.getParameterNode().GetParameter("Segmentation"))
 
+    transformedSegmentationNode = None
+    if not segmentationNode.GetParentTransformNode() is None:
+      # Create a temporary segmentation and harden the transform to ensure that the statistics are calculated
+      # in world coordinates
+      transformedSegmentationNode = slicer.vtkMRMLSegmentationNode()
+      transformedSegmentationNode.Copy(segmentationNode)
+      transformedSegmentationNode.HideFromEditorsOn()
+      slicer.mrmlScene.AddNode(transformedSegmentationNode)
+      transformedSegmentationNode.HardenTransform()
+      self.getParameterNode().SetParameter("Segmentation", transformedSegmentationNode.GetID())
+
```

Comment:
add a try/except block to make sure that we restore the original segmentation node even if an exception is thrown

Reply:
OK, done!